### PR TITLE
gep-19 observability stack - adding fluent-bit operator in logging

### PR DESCRIPTION
**Pull request by nickytd**
/area logging
/area monitoring
/kind enhancement

This PR enhances GEP-19 by adding fluent-bit operator in the logging stack, following the same paradigm shift in monitoring by using public well known operators.

### CODE REVIEW

File: docs/proposals/19-migrating-observability-stack-to-operators.md [hunk 1]
```diff
@@ -1,23 +1,25 @@
 ---
-title:  Monitoring Stack - Migrating to the prometheus-operator
+title:  Observabilits Stack - Migrating to the prometheus-operator and fluent-bit operator
```
Comment (rfranzke) [inline docs/proposals/19-migrating-observability-stack-to-operators.md]: typo

File: docs/proposals/19-migrating-observability-stack-to-operators.md [hunk 2]
```diff
@@ -492,15 +508,126 @@ Add a [sidecar][grafana-sidecar] to Grafana that will pickup dashboards and prov
...
+1. `ClusterInputs`, `ClusterOutputs`, `ClusterFilters` and `ClusterParsers` defining the fluent-but app configuration.
+
+1. `FluentBit` custom resource
+
+```yaml
+apiVersion: fluentbit.fluent.io/v1alpha2
+kind: FluentBit
+metadata:
+  name: fluent-bit
+  namespace: fluent
```
Comment (rfranzke) [inline docs/proposals/19-migrating-observability-stack-to-operators.md]: The namespace should probably stay `garden`?

File: docs/proposals/19-migrating-observability-stack-to-operators.md [hunk 3]
```diff
@@ -492,15 +508,126 @@ Add a [sidecar][grafana-sidecar] to Grafana that will pickup dashboards and prov
...
+    timeKey: time
+    timeFormat: "%d/%b/%Y:%H:%M:%S %z"
+    types: "code:integer size:integer"
+    regex: '^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$'
+
+```
+
+In this example, we have a fluent bit [filter-plugin] of type parser and the corresponding [regex-parser]. In the Gardener context, these configuration resources are supplied by the core components and extensions.
+
+### BYOMC (Bring your own logging configuration)
```
Comment (rfranzke) [inline docs/proposals/19-migrating-observability-stack-to-operators.md]:
```suggestion
### BYOLC (Bring your own logging configuration)
```

File: docs/proposals/19-migrating-observability-stack-to-operators.md [hunk 4]
```diff
@@ -492,15 +508,126 @@ Add a [sidecar][grafana-sidecar] to Grafana that will pickup dashboards and prov
...
+
+Since fluent-bit uses [input-tail] plugin and reads any container output under `/var/log/pods` it processes logs outputs of all workloads. In this case, any workload may bring its own set of filters and parsers if needed using the declarative APIs supported by the operator.
+
 ### Migration
 
+#### Promtheus Operator
```
Comment (rfranzke) [inline docs/proposals/19-migrating-observability-stack-to-operators.md]:
```suggestion
#### Prometheus Operator
```

**Review by rfranzke — COMMENTED** [review]
Thanks!

File: docs/proposals/19-migrating-observability-stack-to-operators.md (hunk 1 above)
Comment (nickytd) [reply docs/proposals/19-migrating-observability-stack-to-operators.md]: thanks

File: docs/proposals/19-migrating-observability-stack-to-operators.md [hunk 5]
```diff
@@ -1,23 +1,25 @@
 ---
```
Comment (rfranzke) [inline docs/proposals/19-migrating-observability-stack-to-operators.md:1]: Adapt the link in https://github.com/gardener/gardener/blob/master/docs/README.md to the new file name

**Review by rfranzke — COMMENTED** [review]
Looks good, minor nit.
/cc @Kristian-ZH

**Review by rfranzke — COMMENTED** [review]
/approve

**Review by Kristian-ZH — APPROVED** [review]
/lgtm

**Pull request merged**